pkg/cmd/cli/secrets/sshauth.go

```diff
@@ -0,0 +1,194 @@
+provide this information by creating a 'sshauth' secret and attaching it to your service account.`
+
+	createSSHAuthSecretExample = `  // If your SSH authentication method requires only private SSh key, add it by using:
+  $ %[1]s SECRET_NAME --ssh-privatekey=FILENAME
+
+  // If your SSH authentication method requires also CA certificate, add it by using:
+  $ %[1]s SECRET_NAME --ssh-privatekey=FILENAME --ca-cert=FILENAME
+
+  // If you do already have a .gitconfig file needed for authentication, you can create a gitconfig secret by using:
+  $ %[2]s SECRET_NAME path/to/.gitconfig`
+)
+
+// CreateSSHAuthSecretOptions holds the credential needed to authenticate against SCM servers.
+type CreateSSHAuthSecretOptions struct {
+	SecretName      string
+	PrivateKeyPath  string
+	CertificatePath string
+	GitConfigPath   string
+
+	PromptForPassword bool
+
+	Out io.Writer
+
+	SecretsInterface client.SecretsInterface
+}
+
+// NewCmdCreateSSHAuthSecret implements the OpenShift cli secrets new-sshauth subcommand
+func NewCmdCreateSSHAuthSecret(name, fullName string, f *kcmdutil.Factory, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSSHAuthSecretOptions{
+		Out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SECRET_NAME --ssh-privatekey=FILENAME [--ca-cert=FILENAME --gitconfig=FILENAME]", name),
+		Short:   "Create a new secret for ssh authentication",
+		Long:    createSSHAuthSecretLong,
+		Example: fmt.Sprintf(createSSHAuthSecretExample, fullName, newSecretFullName, ocEditFullName),
+		Run: func(c *cobra.Command, args []string) {
+			if err := o.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if err := o.Validate(); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(c, err.Error()))
+			}
+
+			if len(kcmdutil.GetFlagString(c, "output")) != 0 {
+				secret, err := o.MakeSSHAuthSecret()
+				kcmdutil.CheckErr(err)
+
+				kcmdutil.CheckErr(f.PrintObject(c, secret, out))
+				return
+			}
+
+			if err := o.CreateSSHAuthSecret(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&o.PrivateKeyPath, "ssh-privatekey", "", "Path to the ssh private key")
+	cmd.Flags().StringVar(&o.CertificatePath, "ca-cert", "", "Path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig", "", "Path to the .gitconfig file")
+
+	// autocompletion hints
+	cmd.MarkFlagFilename("ssh-privatekey")
+	cmd.MarkFlagFilename("ca-cert")
+	cmd.MarkFlagFilename("gitconfig")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+// CreateSSHAuthSecret saves created Secret structure and prints the secret name to the output on success.
+func (o *CreateSSHAuthSecretOptions) CreateSSHAuthSecret() error {
+	secret, err := o.MakeSSHAuthSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(o.GetOut(), "secret/%s\n", secret.Name)
```

Comment:
If we've picked up the latest in upstream there is a new way of outputting this that formats them differently.  Not sure what code it is but it's used in create.go upstream